spec/core/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -589,7 +593,7 @@ function sendPacket(
 
     // check that the channel is not closed to send packets; 
     abortTransactionUnless(channel !== null)
-    abortTransactionUnless(channel.state !== CLOSED)
+    abortTransactionUnless(channel.state !== CLOSED && channel.flushStatus === NOTINFLUSH)
```

Comment:
this was written initially to allow for optimistic sends right? (That's why in spec we do allow `channel.State` to be in states other than `OPEN`, excluding `CLOSED`). 

Currently, in the implementation, we err if the state [isn't `OPEN`](https://github.com/cosmos/ibc-go/blob/b60617df557264abf23b376255f4375bcd8d27eb/modules/core/04-channel/keeper/packet.go#LL36C2-L36C2), wouldn't this cover us during an upgrade? Could we ever find ourselves in a situation where `channel.State == OPEN` and `channel.FlushStatus` is anything but `NOTINFLUSH`?

Reply:
would be nice to add either way to play it as defensively as possible.